include/exadg/convection_diffusion/spatial_discretization/operator.h

```diff
@@ -89,6 +90,12 @@ class Operator : public dealii::Subscriptor, public Interface::Operator<Number>
   void
   setup_solver(double const scaling_factor_mass = -1.0, VectorType const * velocity = nullptr);
 
+  /**
+   * Initializes dealii::DoFHandlers and dealii::AffineConstraints.
+   */
+  void
+  initialize_dof_handler_and_constraints();
```

Comment:
We have to be careful with making this public. This function resets shared pointers that are used in other public functions e.g. in `fill_matrix_free_data()`. Hence, other classes might rely on what has been set in the constructor of `ConvDiff::Operator`.

Reply:
see PR #607

Reply:
I think the function should remain private and we should introduce a new public member function `setup_after_adaptive_mesh_refinement()`, which internally uses as much as possible existing member functions.